### DISCUSSION TITLE
Advantages and disadvantages of setting a large value for modelCacheSize

### DISCUSSION THREAD

**Discussion by futahei** [google/model-viewer#4049] [Q&A] [answered]
Hello, I'm looking for a solution to a problem I've been having with model-viewer.
I was looking for a solution to the slow operation of multiple model-viewers on the same page, and while reading the documentation, I found a static property called modelCacheSize.
According to the description, it seems to be able to control the number of cached models, but I didn't know what advantages or disadvantages there would be in terms of behavior if I set a large value. I would like to know that.
I found a sample that tests this.

https://modelviewer.dev/examples/add

(sorry for the strange English, I'm using a translation tool)

**Comment by elalish** [comment, marked as answer]
It depends what you mean by slow, and it would help to have a link to a demonstration page. All modelCacheSize does is allow GLBs to be garbage-collected when they are not being used by any m-v elements. This will cause them to be re-uploaded to the GPU if you decide to use that model again, rather than being instantly available (the browser likely caches the file already). The only slowness it could affect is that of switching between models. 

If you're having trouble with frame-rate, then you should look at optimizing your models (reducing the number of materials, the size of the textures and the number of triangles).
